modules/caddyhttp/reverseproxy/caddyfile.go

```diff
@@ -104,6 +104,13 @@ func (h *Handler) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 		var network, scheme, host, port string
 
 		if strings.Contains(upstreamAddr, "://") {
+			// we get a parsing error if a placeholder is specified
+			// so we return a more user-friendly error message instead
+			// to explain what to do instead
+			if strings.Contains(upstreamAddr, "{") {
+				return "", d.Err("for now, placeholders are not allowed when specifying a scheme; instead, omit the scheme and use the transport subdirective if you need tls support")
```

Comment:
This does assume that the transport is capable of TLS -- I guess that's a mostly reasonable assumption, but some aren't, for example I don't think the fastcgi transport can do TLS.

Maybe just something like:
```suggestion
				return "", d.Err("due to parsing difficulties, placeholders are not allowed when an upstream address contains a scheme")
```